ebos/eclproblem.hh

```diff
@@ -1716,7 +1694,9 @@ public:
     {
         OPM_TIMEBLOCK_LOCAL(wellTransMultiplier);
         
-        double trans_mult = this->simulator().problem().template rockCompTransMultiplier<double>(intQuants, elementIdx);
+        bool implicit = !EWOMS_GET_PARAM(TypeTag, bool, ExplicitRockCompaction);
+        double trans_mult = implicit? this->simulator().problem().template computeRockCompTransMultiplier_<double>(intQuants, elementIdx) :
+                            this->simulator().problem().getRockCompTransMultVal(elementIdx);
         trans_mult *= this->simulator().problem().template permFactTransMultiplier<double>(intQuants);
```

Comment:
This is still implicit, should this factor also be explicit if RockCompaction is?

If keeping as is, you could simplify with
```
double trans_mult = rockCompTransMultiplier<double>(intQuants, elementIdx);
```
which already contain the above logic.

Reply:
This is still implicit, should this factor also be explicit if RockCompaction is?

I dont think that is necessary.  The perfFactTransMultiplier is related to salt precipitation and are different from rock compaction.